src/main/flight/pid.c

```diff
@@ -869,9 +877,58 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, timeUs_t currentTim
     rpmFilterUpdate();
 #endif
 
+#ifdef USE_CHIRP
+
+    static int chirpAxis = 0;
+    static bool shouldChirpAxisToggle = true;
+
+    float chirp = 0.0f;
+    float sinarg = 0.0f;
+    if (FLIGHT_MODE(CHIRP_MODE)) {
+
+        // toggle chirp signal logic
+        if (shouldChirpAxisToggle) {
+            shouldChirpAxisToggle = false;
+        }
+
+        // update chirp signal
+        if (chirpUpdate(&pidRuntime.chirp)) {
+            chirp = pidRuntime.chirp.exc;
+            sinarg = pidRuntime.chirp.sinarg;
+        }
+    } else {
+        if (!shouldChirpAxisToggle) {
+
+            // toggle chirp signal logic and increment to next axis for next run
+            shouldChirpAxisToggle = true;
+            chirpAxis++;
+            if (chirpAxis > FD_YAW) {
+                chirpAxis = 0;
+            }
+
+            // reset chirp signal generator
+            chirpReset(&pidRuntime.chirp);
+        }
+    }
+
+    // input / excitation shaping
+    float chirpFiltered  = phaseCompApply(&pidRuntime.chirpFilter, chirp);
+
+    // many tools use debug gyro scaled
+    DEBUG_SET(DEBUG_GYRO_SCALED, 3, lrintf(5.0e3f * sinarg));
```

Comment:
Another option is to emit 'chirp start' blackbox event. When analyzing, you can regenerate chirp locally, saving debug for other purposes. Chirp generations seems numerically stable, so it should be quite easy to recreate the chirp with enough precision.

Reply:
Nice! Thank you for the info about ``_approx`` accuracy. I will put this on my ToDo list and test it! First i will change the debug though since i saw a PR where ``GYRO_FILTERED`` will be removed.

Reply:
Ok, only one debug CHIRP left. Will test if i can remove it as well.

Reply:
pretty rainy here, hope i can test this any weekend soon

Reply:
I would like to leave it for now.